src/Quaternion.jl

```diff
@@ -193,10 +193,13 @@ function qrotation(axis::Vector{T}, theta) where {T <: Real}
     if length(axis) != 3
         error("Must be a 3-vector")
     end
-    u = normalize(axis)
-    thetaT = convert(eltype(u), theta)
-    s = sin(thetaT / 2)
-    Quaternion(cos(thetaT / 2), s * u[1], s * u[2], s * u[3], true)
+    normaxis = norm(axis)
+    if iszero(normaxis)
+        error("must provide nonzero rotation axis")
+    end
```

Comment:
Why this change in behavior? Currently if one provides a zero axis, i.e. corresponding to no rotation at all, `qrotation` returns `Quaternion(1)`, the multiplicative identity, which seems correct.

Reply:
I'd prefer you open an issue to discuss it. This would be a breaking change (the current behavior is not a bug), and I don't want to hold up the main fix from this PR being merged.

Reply:
I'll leave it alone. It would indeed be disruptive to throw an error where one was not thrown previously. As suggested, I've changed the computation to instead force the rotation to zero when the axis is zero.